### ISSUE TITLE
iOS9: unable to discover airplay devices created by airsonos

### ISSUE THREAD

**Issue by SteffenCL** [stephen/airsonos#177]
With the current beta of iOS9 the airplay devices created by airsonos are not discoverable from iOS9 devices. 

From my iPad which is still running iOS8 the airplay devices are discoverable.

As iOS9 is not yet released for public use this might be fixed with later builds. But wanted to let you know about it anyway.

Kind regards,
Steffen

**Comment by mhamoser** [comment]
I can confirm the the devices aren't found in iOS9. In addition they are not found in OS X El Capitan in the general output settings either. They are found in iTunes in El Capitan, but can't be used due to an error 1500 (airtunes and iTunes codecs can't be used e.g. apple lossless).

**Comment by tearsen** [comment]
I can also confirm, on iOS 9 beta1 the airplay devices aren´t found.

**Comment by alijnclarke** [comment]
+1, neither iOS 9 or 10.11 work

**Comment by rodneyyziek** [comment]
Issue remains in beta 2 (both iOS and OS X)

**Comment by jonarribasmateo** [comment]
+1

**Comment by spliket** [comment]
Issue remains in beta 3 on iOS 9.

**Comment by gzeel** [comment]
Same problem over here with iOS 9 beta 3. I can confirm that it still works on iOS 8.4

**Comment by wcm4u** [comment]
I confirm iOS 9 public Beta don't see any Speakers from airsonos. But the Mac Software SonoAir also fails, no Speakers shown up on iOS9 Public beta.

**Comment by RToo** [comment]
iOS 9, Incompatible Old AirPlay Protocols.

**Comment by kschio** [comment]
I can confirm this issue as well, would be great if airsonos would be compatible with OS X 10.11.

**Comment by pdxsam** [comment]
I also can confirm. Not seen on IOS9 beta

**Comment by gloos** [comment]
Is this going to be fixed? I can't see it on ios 9 beta either.

**Comment by wcm4u** [comment]
I think apple has to fix ...

-- 
Web Content Management[4u]
Ramon Messerschmidt
Anwendungs- und Webentwickler

Hasenweg 11
33790 Halle (Westf.)

fon: 05201-6580026
fax: 05201-156836-9
mobile: 0176-62796955
e-mail:  rm@wcm4u.de
web: www.wcm4u.de

**Comment by spliket** [comment]
@wcm4u, Apple manufactured devices (Apple TV and others) show up as Airplay devices in the beta. I don't think this is an issue with their Airplay protocol. Obviously, they changed something, but their devices work. So, I doubt a fix will come from them.

**Comment by pdxsam** [comment]
Sadly, you also can't go back from IOS9.  You can reload the backup  but IOS9 changes the firmware on the iPad and even if you go back, you can't see Airsonos.

I wonder if it has to do with the warning that comes up when starting Airsonos about Bonjour. Apple must have updated their Bonjour code and it's causing IOS to miss the Airsonos advertisements.

**Comment by pdxsam** [comment]
Oops  update..  you can go back. I just did it incorrect. I still think it might be Bonjour.

**Comment by stephen** [comment]
Given [reports](http://appleinsider.com/articles/15/07/08/apple-issues-third-betas-of-ios-9-watch-os-os-x-1011-el-capitan-to-developers) that AirPlay is spotty in both betas, this likely won't be resolved until both are GA.

**Comment by hirvonen** [comment]
Issue remains in beta 5 on iOS 9.
I think apple has changed the encrypt way of Airplay to strengthen security or something.

**Comment by gloos** [comment]
I also confirm it still doesn't work in in the latest beta.

**Comment by stephen** [comment]
Any further information for debugging this issue would be nice. Otherwise, this won't get investigated until iOS 9 hits GA.

- Do other AirPlay (AirPort Express, AppleTV) devices show up?
- Does the bonjour advertisement show up at all according to the iOS device? https://itunes.apple.com/us/app/discovery-bonjour-browser/id305441017?mt=8

**Comment by jonarribasmateo** [comment]
**Do other AirPlay (AirPort Express, AppleTV) devices show up?**

@stephen Sure it does, I use air server (www.airserver.com) and it works like a charm.
![img_0238](https://cloud.githubusercontent.com/assets/4953393/9698471/f5ba4838-53b7-11e5-91d0-e7b2e7d8d38a.PNG)

**Does the bonjour advertisement show up at all according to the iOS device?**
_raop._tpc.  OK  (although there is no option as sound output). Not listed in AirPlay.

![image1](https://cloud.githubusercontent.com/assets/4953393/9698491/fec26e82-53b8-11e5-9941-e2f204cf749d.PNG)
 

545200C2C99A@Salón (SonoAir)._raop._tcp.local.
Remote Audio Output Protocol (AirTunes)

Mac-mini-de-Jon.local:5003
192.168.1.129:5003
fe80::aa20:66ff:fe11:c4cf%en0:5003

am = Salón (SonoAir)
ch = 2
cn = 0
ek = 1
et = 0,1
md = 0
pw = false
sr = 44100
ss = 16
tp = TCP,UDP
tx = 1
txtvers = 1
vs = 130.14

deviceid = 20:C9:D0:90:EB:27
features = 0x100029ff
model = AppleTV3,2
rhd = 5.3.2
srcvers = 150.33
vv = 2

**Comment by Easen** [comment]
Hi @stephen, it seems if you include the 'vn=65537' parameter in the mDNS advertisement in Nodetunes, AirSonos appears in iOS9. However the audio feed that is transmitted is ALAC.

**Comment by t3chn0m4g3** [comment]
Hello @stephen, same problem here, my iOS 8.x devices can discover, my iOS 9 GA devices unfortunately can not. Happy to help and support!

**Comment by m47een** [comment]
Hello @stephen. Just to confirm after upgrading to iOS 9 GA airsonos doesn't seem to be working. Happy to help with any testing. Thanks

**Comment by xonbul** [comment]
HI @stephen, I can confirm the problem the problem as well. Can't discover sonos after upgrade to iOS9. Ok with iOS8 devices and Mac OSX 10.10.5

**Comment by MrSimmo** [comment]
Same here, ios 9 gm and ga fail to discover airsonos across iphone and ipads

**Comment by hansmestrum** [comment]
Same here. updated to iOS9 and my sonos speakers are not visible on my iphone. Hope this will be solved soon as arises is great. I also use my old turntable to play vinyl via airsonos to my sonos.
I also read about problems with El Capitan. Hope this will also be solved. I do not want to loose airsonos.

**Comment by matthijst** [comment]
I'm sure Stephen is aware of the issue by now? Or do you really think he needs 20 more posts from people confirming Airsonos doesn't work on their iOS9 device anymore either?

**Comment by MrSimmo** [comment]
A larger number of people reporting the problem demonstrates that it is not user error.

**Comment by gzeel** [comment]
Can we please stop with these 'confirm' messages? Looks to me it's an airplay protocol issue, so every user that has upgraded to iOS 9 has this problem.

**Comment by hansmestrum** [comment]
Right. Would be nice to hear if someone is working on it.

**Comment by Easen** [comment]
There's two issues here, 
1) Missing parameter from the mDNS (see my previous comment). 
2) The lack of ALAC support, see #1.

**Comment by MartinAyla** [comment]
I hope it's technically possible to fix AirSonos for iOS 9 - crossing fingers.

**Comment by hansmestrum** [comment]
yeah and for El Captian later this month. Otherwise I loose all airsonos :-(

**Comment by shining01** [comment]
i hope too. i love airsonos. i hope you can find the problem and fix it. thanks

**Comment by stephen** [comment]
Thanks all for the reports. It looks like Apple has dropped support for devices that don't transmit audio via at least ALAC in iOS 9 (and presumably El Capitan).

Tracking the issue in nodetunes here: https://github.com/stephen/nodetunes/issues/1
